### DISCUSSION TITLE
How do I add additional labels and additional Env variables for mater nodes

### DISCUSSION THREAD

**Discussion by NoorKumar** [Opster/opensearch-k8s-operator#133] [Q&A] [answered]
Hi Team,
I was trying to create the cluster using operator, I do not see the way to add any additional labels or any extra environment variables to any nodes (master nodes or data nodes). When I submit the cluster config file to operator to create cluster I would like add some labels and env variables. Please suggest if there are any ways to include those ?

**Comment by prudhvigodithi** [comment, marked as answer]
Hey @NoorKumar `additional labels` and `extra environment variables` here means you are trying add labels and vars to the created kubernetes components by the operator? if so right now operator manages the labels with `clusterName` and `componentName`, I see you have an [issue](https://github.com/Opster/opensearch-k8s-operator/issues/135) already created for this which I can take a look.

**Reply by NoorKumar** [reply]
Hi @prudhvigodithi 

`apiVersion: opensearch.opster.io/v1
kind: OpenSearchCluster
metadata:
  name: my-cluster
  namespace: opensearch-demo
spec:
  general:
    version: 1.3.0
    httpPort: 9200
    vendor: opensearch
    serviceName: my-cluster
  dashboards:
    version: 1.3.0
    enable: true
    replicas: 2
    resources:
      requests:
         memory: "1Gi"
         cpu: "500m"
      limits:
         memory: "1Gi"
         cpu: "500m"
  confMgmt:
    smartScaler: true
  nodePools:
    - component: masters
      replicas: 3
      # here I want to add my own labels along with whatever operator is adding, and env vars to the opensearch container. Operator should those labels and env vars to all the master nodes. 
      diskSize: "10Gi"
      NodeSelector:
      resources:
         requests:
            memory: "2Gi"
            cpu: "500m"
         limits:
            memory: "2Gi"
            cpu: "500m"
      roles:
        - "master"
    - component: nodes
      replicas: 2  # here I want to add my own labels along with whatever operator is adding, and env vars to the opensearch container. Operator should those labels and env vars to all the data nodes. 
      diskSize: "10Gi"
      NodeSelector:
      resources:
         requests:
            memory: "2Gi"
            cpu: "500m"
         limits:
            memory: "2Gi"
            cpu: "500m"
      roles:
        - "data"`

If you look the above config file (the one which given in example), I do not have any way to add labels or env vars. I want to add few labels to master node and data node as well and operator can take those labels and operator can add it's own labels (whatever adding as of today). Same for Env vars, I will be using those env vars and labels later.

**Reply by treknado** [reply]
@NoorKumar If it is still needed
```
spec:
  nodePools:
    - component: masters
      replicas: 3

      labels:
        app: Opensearch #example

      env:
        - name: MY_ENV_VAR
          value: "myvalue"

      diskSize: "10Gi"
```
best regards,
Treknado
